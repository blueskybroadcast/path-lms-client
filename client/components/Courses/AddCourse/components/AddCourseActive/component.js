import React from 'react';
import PropTypes from 'prop-types';

const AddCourseActive = ({ active, handleChange }) => (
  <fieldset>
    <legend>
      <span>Active</span>
    </legend>
    <div className="control-group active-toggle">
      <label htmlFor="active">Active</label>
      <div className="controls">
        <label
          className="label-switch"
          tabIndex="0"
          htmlFor="active"
        >
          <input
            className="boolean optional switcher optional "
            id="active"
            type="checkbox"
            onChange={handleChange}
            checked={active}
            name="active"
          />
          <div className="checkbox" />
        </label>
        <p className="help-block">Users can view this content.</p>
      </div>
    </div>
  </fieldset>
);

AddCourseActive.propTypes = {
  active: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default AddCourseActive;
