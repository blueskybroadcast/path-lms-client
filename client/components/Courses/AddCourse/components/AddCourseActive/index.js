import AddCourseActive from './component';

export default AddCourseActive;
