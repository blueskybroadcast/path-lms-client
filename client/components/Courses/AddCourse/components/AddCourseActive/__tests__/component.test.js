import React from 'react';
import { shallow } from 'enzyme';
import AddCourseActive from '../component';

test('should render AddCourseActive correctly', () => {
  const wrapper = shallow(
    <AddCourseActive
      active
      handleChange={jest.fn()}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
