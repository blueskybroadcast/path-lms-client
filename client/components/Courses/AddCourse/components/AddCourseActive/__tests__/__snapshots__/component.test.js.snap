// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddCourseActive correctly 1`] = `
<fieldset>
  <legend>
    <span>
      Active
    </span>
  </legend>
  <div
    className="control-group active-toggle"
  >
    <label
      htmlFor="active"
    >
      Active
    </label>
    <div
      className="controls"
    >
      <label
        className="label-switch"
        htmlFor="active"
        tabIndex="0"
      >
        <input
          checked={true}
          className="boolean optional switcher optional "
          id="active"
          name="active"
          onChange={[MockFunction]}
          type="checkbox"
        />
        <div
          className="checkbox"
        />
      </label>
      <p
        className="help-block"
      >
        Users can view this content.
      </p>
    </div>
  </div>
</fieldset>
`;
